server/src/main/java/com/hedera/block/server/BlockNodeAppInjectionComponent.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server;
+
+import com.hedera.block.server.health.HealthInjectionModule;
+import dagger.Component;
+import javax.inject.Singleton;
+
+/** The infrastructure used to manage the instances and inject them using Dagger */
+@Singleton
+@Component(
+        modules = {
+            BlockNodeAppInjectionModule.class,
+            HealthInjectionModule.class,
+        })
```

Comment:
Short answer No.
Long answer Maybe...

as we currently had the Configuration is created as part of the 'BlockNodeContext' so we don't really need to create a config injection module, and the BlockNodeContext is being injected as part of the BlockNodeInjectionModule (that also contains the metricsService)

So I would say that now that we have refactored for DI we might want to reconsider how we are creating the config and metrics as part of the BlockNodeContext, and maybe ever get rid of it and let DI inject one or the other where needed.

However I would suggest that to be part of a follow up PR, not even DI Dagger Part 2. I preffer to avoid big refactors when possible. (we can create a ticket for this.)

Reply:
Very reasonable to defer this for a separate issue.
I would advocate for using the platform config and logging sooner rather than later to minimize the amount of change needed to make this system work with existing devops processes and expectations.

Reply:
Fair, I've created this ticket to follow up on this after `Part 2` of Dagger DI Refactor is completed.
https://github.com/hashgraph/hedera-block-node/issues/126